src/components/Aframe/my-loader.js

```diff
@@ -298,130 +298,118 @@ AFRAME.registerComponent("myloader", {
     }
   },
 
-  loadModel: function () {
-    var currentVolume = this.el.getObject3D("mesh");
-    if (currentVolume !== undefined) {
-      //clear mesh
-      currentVolume.geometry.dispose();
-      currentVolume.material.dispose();
-      this.el.removeObject3D("mesh");
-      this.el.sceneEl.object3D.dispose();
-      currentVolume = undefined;
-    }
+  loadModel: function (fullPath, volumeProperties) {
+
+	  var currentVolume = this.el.getObject3D('mesh');
+	  if (currentVolume !== undefined) {
+			//clear mesh
+		  currentVolume.geometry.dispose();
+		  currentVolume.material.dispose();
+		  this.el.removeObject3D('mesh');
+		  this.el.sceneEl.object3D.dispose();
+		  currentVolume = undefined;
+		}
+
+	   if (fullPath !== "") {
+		  this.hiddenLabel.style.display = '';
+		  var el = this.el;
+		  var data = this.data;		
+		  var myWidth = this.myCanvas.width;
+		  var myheight = this.myCanvas.height;
+		  var colorMap = null;
+		  var useTransferFunction;
+		  var hiddenLabel = this.hiddenLabel;
+		  var enabledColorMapping = this.colorMapEnabled;
+		  var iam = this;
```

Comment:
what is the need to remap this? what does `iam` stand for?

Reply:
Should we also discuss using `let` instead of `var` to keep scoping more clear/safer?

Reply:
probably.  many of these should probably be `consts`, but I'm not fully sure which

Reply:
@kmilo9999 Are comfortable giving a pass and replacing `var` for `let` and `const` instead? Or would it be helpful to have a short discussion to clarify the differences between them?

Reply:
In general the basic consensus is

> Use let when you know that the value of a variable will change.
> Use const for every other variable.
> Do not use var.

The scope of `var` can introduce more bugs - short description/summary [here](https://codeburst.io/javascript-var-let-or-const-which-one-should-you-use-2fd521b050fa)

Reply:
I remember why I did that. 
So, the previous implementation of THREE.TextureLoader().load only had two parameters. the texture file path and a function promise to execute after the texture was allocated in memory. I coded all the operations of binding the  texture to the shader in this promise function. The problem is that the reference to **this** doesnt exist inside the execution of the promise. The workaround I found was loading variabales on stack (the remaping you mentioned) and then retrieve them inside the function (javascript magic).
I just took a look a the new THREE.TextureLoader().load, and it seems now it doesnt need this promise anymore (although, still supports it). 
If the remaping is such an inconvenience, I can just recode that part again without the promise.
The only const value I see is `uniforms["step_size"].value` .

Reply:
I don't think that rewrite is important right now (maybe a follow on PR, but let's get things working first).  I think what is more important is making the code more readable/maintainable in place where we can.  So maybe add a comment with this info?  I also wonder if `iam` could be named something better or if it would be better to call out the variables used into variables so that they're available in that function's scope.

Reply:
As far as `var` vs `let` vs `const`, I'm doing a solid chunk of general cleanup in a separate branch. I can do a sweep of the variable scoping there if no one objects

Reply:
you are right @mcmcgrath13 . iam is a terrible variable name. what about pre appending the word **promise** to the name of the variable?
`var promiseThis`
or
`var promiseMyLoader`

The same goes for other variables.

I dont have any issues changing `var` to `let`. I like standardizing the code across projects.

Reply:
It looks like the only place `iam` is used is

Reply:
I applied your last suggestion Mary.